spec/api-callbacks-registry-spec.js

```diff
@@ -0,0 +1,42 @@
+const assert = require('assert')
+const {CallbacksRegistry} = require('electron')
+
+describe('CallbacksRegistry module', () => {
+  let registry = null
+
+  beforeEach(() => {
+    registry = new CallbacksRegistry()
+  })
+
+  it('adds a callback to the registry', () => {
+    const cb = () => [1, 2, 3, 4, 5]
+    const id = registry.add(cb)
+    assert.equal(id, 1)
+  })
+
+  it('returns a specified callback if it is in the registry', () => {
+    const cb = () => [1, 2, 3, 4, 5]
+    registry.add(cb)
+
+    const callback = registry.get(1)
+    assert.equal(callback.toString(), cb.toString())
+  })
+
+  it('returns an empty function if the cb doesnt exist', () => {
+    const callback = registry.get(1)
+    assert.equal(callback.toString(), 'function () {}')
```

Comment:
I think this is again testing the implementation details a little too closely.

We want to know that it returns _something_ for an invalid key, but it's not intrinsically an empty function, it could be one that logs a warning or whatever.

Better: `assert.isFunction(callback)`

Reply:
oh i can do that?? rad...TIL!